apps/homework/templates/homework/submission_list.html

```diff
@@ -87,7 +88,13 @@ <h4>Due Date: {{ definition.due_date }}</h4>
                                                             {% endif %}
                                                             <td>
                                                                 {% if submission.github_url %}
-                                                                    <a class="ui mini blue icon button" href="{{ submission.github_url }}" target="_blank" rel="noopener noreferrer">
+                                                                    <a class="ui mini blue icon button"
+                                                                        {% if definition.jupyter_notebook_enabled  %}
+                                                                            href="{{ submission.github_url|nb_viewer_format_for_github_jupyter_notebook_url }}"
+                                                                        {% else %}
+                                                                            href="{{ submission.github_url }}"
+                                                                        {% endif %}
+                                                                       target="_blank" rel="noopener noreferrer">
```

Comment:
I feel like this would read a bit nicer if there was a property on the submission model that returned either github url or notebook url? in general moving logic out of templates is ideal, but can't always do it!

Reply:
Model: **big**
Template: *smol*